java/src/Ice/src/main/java/com/zeroc/Ice/Communicator.java

```diff
@@ -110,7 +110,8 @@ public boolean isShutdown() {
    * @see #proxyToString
    */
   public ObjectPrx stringToProxy(String str) {
-    return _instance.proxyFactory().stringToProxy(str);
+    var ref = _instance.referenceFactory().create(str, null);
+    return (ref == null) ? null : new com.zeroc.Ice._ObjectPrxI(ref);
   }
```

Comment:
This was the only place where these 3 `proxyFactory` methods were being called.
So I decided to just inline the logic here, and delete them out of `proxyFactory`.

The implementations are pretty simple anyways.